The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5519902111/jobs/10065789864) [(plain)](https://github.com/rust-lang/rust/commit/3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce/checks/14944704681/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
  lfs: false
  submodules: false
  set-safe-directory: true
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/should-skip-this.sh
src/ci/scripts/should-skip-this.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-channel.sh
src/ci/scripts/verify-channel.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/collect-cpu-stats.sh
src/ci/scripts/collect-cpu-stats.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-sccache.sh
src/ci/scripts/install-sccache.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/select-xcode.sh
src/ci/scripts/select-xcode.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/install-clang.sh
src/ci/scripts/install-clang.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/install-wix.sh
src/ci/scripts/install-wix.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-msys2.sh
src/ci/scripts/install-msys2.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/install-mingw.sh
src/ci/scripts/install-mingw.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/install-ninja.sh
src/ci/scripts/install-ninja.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/enable-docker-ipv6.sh
src/ci/scripts/enable-docker-ipv6.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/verify-line-endings.sh
src/ci/scripts/verify-line-endings.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-tools
##[endgroup]
##[group]Run src/ci/scripts/verify-backported-commits.sh
src/ci/scripts/verify-backported-commits.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-stable-version-number.sh
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/run-build-from-ci.sh
src/ci/scripts/run-build-from-ci.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 3fbb9a02dd9c8562ab93d189c980fcd05e3d37ce
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
tests/ui/crashes/ice-10148.rs ... ok
tests/ui/missing_const_for_fn/cant_be_const.rs ... ok

tests/ui/empty_line_after_doc_comments.rs FAILED:
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Zui-testing" "-Dwarnings" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rlib" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rmeta" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rlib" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rmeta" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libderive_new-c02f7361dc6f59be.so" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rmeta" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libitertools-0f51f8483eab9879.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libitertools-0f51f8483eab9879.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rmeta" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/libclippy_test_deps.rlib" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_test_deps-728ed027f0623fde.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/thiserror-60234973db2aebdd" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/memchr-300f0ecb3b085bd3" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/libc-467a9bf5fecaa5b9" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/pulldown-cmark-5d2703cd75133d86" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/semver-367f0564b96a3f38" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-task-14f797a867cbbbea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/tokio-988b1ca3d2062fb8" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/syn-c0e5e037aa06912e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde_json-06fa4ab4a6a2c54b" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-channel-18c61a4c728f3f63" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/quote-fc00b89c3984f431" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-core-67d40d21860bf7e2" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde-b5ab3846bee6426e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/camino-95d981076bbda069" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/proc-macro2-c56f6977d10d30ea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/slab-084e9e34eeb9970c" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/lock_api-172e9730276f85f1" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/unicase-dc66fd2ee33a4c53" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/parking_lot_core-ff12b1c9a236dc80" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-util-d30abe96b847bc06" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/tests/ui/empty_line_after_doc_comments" "tests/ui/auxiliary/proc_macro_attr.rs" "--edition" "2021" "--crate-type" "proc-macro" "--emit=link"


Aux build from tests/ui/empty_line_after_doc_comments.rs:1 failed
compilation of aux build failed failed with exit status: 1
compilation of aux build failed failed with exit status: 1


full stderr:
error[E0308]: mismatched types
##[error]  --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________expected `Span`, found `DelimSpan`
   |
   = note: this error originates in the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0277]: the trait bound `syn::Signature: quote::ToTokens` is not satisfied
   |
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Signature`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&syn::Signature` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Block: quote::ToTokens` is not satisfied
   |
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Block`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&mut syn::Block` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `quote::__private::TokenStream: syn::__private::ToTokens` is not satisfied
   |
   |
LL |                 *block = parse_quote!(#my_block);
   |                          |
   |                          |
   |                          the trait `syn::__private::ToTokens` is not implemented for `quote::__private::TokenStream`
   |
   |
   = help: the following other types implement trait `syn::__private::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 298 others
   = note: this error originates in the macro `$crate::__private::quote::quote` which comes from the expansion of the macro `parse_quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemTrait: quote::ToTokens` is not satisfied
   |
   |
LL |     TokenStream::from(quote!(#item))
   |                       |
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemTrait`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemImpl: quote::ToTokens` is not satisfied
   |
   |
LL |     TokenStream::from(quote!(#item))
   |                       |
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemImpl`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
---



tests/ui/empty_line_after_outer_attribute.rs FAILED:
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Zui-testing" "-Dwarnings" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rlib" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rmeta" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rlib" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rmeta" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libderive_new-c02f7361dc6f59be.so" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rmeta" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libitertools-0f51f8483eab9879.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libitertools-0f51f8483eab9879.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rmeta" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/libclippy_test_deps.rlib" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_test_deps-728ed027f0623fde.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/thiserror-60234973db2aebdd" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/memchr-300f0ecb3b085bd3" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/libc-467a9bf5fecaa5b9" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/pulldown-cmark-5d2703cd75133d86" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/semver-367f0564b96a3f38" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-task-14f797a867cbbbea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/tokio-988b1ca3d2062fb8" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/syn-c0e5e037aa06912e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde_json-06fa4ab4a6a2c54b" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-channel-18c61a4c728f3f63" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/quote-fc00b89c3984f431" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-core-67d40d21860bf7e2" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde-b5ab3846bee6426e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/camino-95d981076bbda069" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/proc-macro2-c56f6977d10d30ea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/slab-084e9e34eeb9970c" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/lock_api-172e9730276f85f1" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/unicase-dc66fd2ee33a4c53" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/parking_lot_core-ff12b1c9a236dc80" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-util-d30abe96b847bc06" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/tests/ui/empty_line_after_outer_attribute" "tests/ui/auxiliary/proc_macro_attr.rs" "--edition" "2021" "--crate-type" "proc-macro" "--emit=link"


Aux build from tests/ui/empty_line_after_outer_attribute.rs:1 failed
compilation of aux build failed failed with exit status: 1
compilation of aux build failed failed with exit status: 1


full stderr:
error[E0308]: mismatched types
##[error]  --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________expected `Span`, found `DelimSpan`
   |
   = note: this error originates in the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0277]: the trait bound `syn::Signature: quote::ToTokens` is not satisfied
   |
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Signature`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&syn::Signature` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Block: quote::ToTokens` is not satisfied
   |
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Block`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&mut syn::Block` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `quote::__private::TokenStream: syn::__private::ToTokens` is not satisfied
   |
   |
LL |                 *block = parse_quote!(#my_block);
   |                          |
   |                          |
   |                          the trait `syn::__private::ToTokens` is not implemented for `quote::__private::TokenStream`
   |
   |
   = help: the following other types implement trait `syn::__private::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 298 others
   = note: this error originates in the macro `$crate::__private::quote::quote` which comes from the expansion of the macro `parse_quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemTrait: quote::ToTokens` is not satisfied
   |
   |
LL |     TokenStream::from(quote!(#item))
   |                       |
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemTrait`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemImpl: quote::ToTokens` is not satisfied
   |
   |
LL |     TokenStream::from(quote!(#item))
   |                       |
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemImpl`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
---



tests/ui/needless_arbitrary_self_type_unfixable.rs FAILED:
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Zui-testing" "-Dwarnings" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rlib" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rmeta" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rlib" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rmeta" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libderive_new-c02f7361dc6f59be.so" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rmeta" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libitertools-0f51f8483eab9879.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libitertools-0f51f8483eab9879.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rmeta" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/libclippy_test_deps.rlib" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_test_deps-728ed027f0623fde.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/thiserror-60234973db2aebdd" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/memchr-300f0ecb3b085bd3" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/libc-467a9bf5fecaa5b9" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/pulldown-cmark-5d2703cd75133d86" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/semver-367f0564b96a3f38" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-task-14f797a867cbbbea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/tokio-988b1ca3d2062fb8" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/syn-c0e5e037aa06912e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde_json-06fa4ab4a6a2c54b" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-channel-18c61a4c728f3f63" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/quote-fc00b89c3984f431" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-core-67d40d21860bf7e2" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde-b5ab3846bee6426e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/camino-95d981076bbda069" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/proc-macro2-c56f6977d10d30ea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/slab-084e9e34eeb9970c" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/lock_api-172e9730276f85f1" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/unicase-dc66fd2ee33a4c53" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/parking_lot_core-ff12b1c9a236dc80" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-util-d30abe96b847bc06" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/tests/ui/needless_arbitrary_self_type_unfixable" "tests/ui/auxiliary/proc_macro_attr.rs" "--edition" "2021" "--crate-type" "proc-macro" "--emit=link"


Aux build from tests/ui/needless_arbitrary_self_type_unfixable.rs:1 failed
compilation of aux build failed failed with exit status: 1
compilation of aux build failed failed with exit status: 1


full stderr:
error[E0308]: mismatched types
##[error]  --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________expected `Span`, found `DelimSpan`
   |
   = note: this error originates in the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0277]: the trait bound `syn::Signature: quote::ToTokens` is not satisfied
   |
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Signature`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&syn::Signature` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Block: quote::ToTokens` is not satisfied
   |
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
   | |                  ^
   | |                  |
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Block`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&mut syn::Block` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `quote::__private::TokenStream: syn::__private::ToTokens` is not satisfied
   |
   |
LL |                 *block = parse_quote!(#my_block);
   |                          |
   |                          |
   |                          the trait `syn::__private::ToTokens` is not implemented for `quote::__private::TokenStream`
   |
   |
   = help: the following other types implement trait `syn::__private::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 298 others
   = note: this error originates in the macro `$crate::__private::quote::quote` which comes from the expansion of the macro `parse_quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemTrait: quote::ToTokens` is not satisfied
   |
   |
LL |     TokenStream::from(quote!(#item))
   |                       |
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemTrait`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemImpl: quote::ToTokens` is not satisfied
   |
   |
LL |     TokenStream::from(quote!(#item))
   |                       |
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemImpl`
   |
   |
   = help: the following other types implement trait `quote::ToTokens`:
             char
             isize
             i8
             i16
```

</details>

--- a/compiler/rustc_middle/src/traits/solve/inspect.rs
+++ b/compiler/rustc_middle/src/traits/solve/inspect.rs
@@ -32,7 +32,7 @@ pub enum GoalEvaluationKind<'tcx> {
 }
 impl Debug for GoalEvaluation<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        ProofTreeFormatter { f, on_newline: true }.format_goal_evaluation(self)
+        ProofTreeFormatter::new(f).format_goal_evaluation(self)
     }
 }
 
@@ -43,7 +43,7 @@ pub struct AddedGoalsEvaluation<'tcx> {
 }
 impl Debug for AddedGoalsEvaluation<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        ProofTreeFormatter { f, on_newline: true }.format_nested_goal_evaluation(self)
+        ProofTreeFormatter::new(f).format_nested_goal_evaluation(self)
     }
 }
 
@@ -58,7 +58,7 @@ pub struct GoalEvaluationStep<'tcx> {
 }
 impl Debug for GoalEvaluationStep<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        ProofTreeFormatter { f, on_newline: true }.format_evaluation_step(self)
+        ProofTreeFormatter::new(f).format_evaluation_step(self)
     }
 }
 
@@ -78,6 +78,6 @@ pub enum CandidateKind<'tcx> {
 }
 impl Debug for GoalCandidate<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        ProofTreeFormatter { f, on_newline: true }.format_candidate(self)
+        ProofTreeFormatter::new(f).format_candidate(self)
     }
 }

--- a/compiler/rustc_middle/src/traits/solve/inspect/format.rs
+++ b/compiler/rustc_middle/src/traits/solve/inspect/format.rs
@@ -1,11 +1,19 @@
 use super::*;
 
 pub(super) struct ProofTreeFormatter<'a, 'b> {
-    pub(super) f: &'a mut (dyn Write + 'b),
-    pub(super) on_newline: bool,
+    f: &'a mut (dyn Write + 'b),
 }
 
-impl Write for ProofTreeFormatter<'_, '_> {
+/// A formatter which adds 4 spaces of identation to its input before
+/// passing it on to its nested formatter.
+///
+/// We can use this for arbitrary levels of indentation by nesting it.
+struct Indentor<'a, 'b> {
+    f: &'a mut (dyn Write + 'b),
+    on_newline: bool,
+}
+
+impl Write for Indentor<'_, '_> {
     fn write_str(&mut self, s: &str) -> std::fmt::Result {
         for line in s.split_inclusive("\n") {
             if self.on_newline {
@@ -19,49 +27,52 @@ impl Write for ProofTreeFormatter<'_, '_> {
     }
 }
 
-impl ProofTreeFormatter<'_, '_> {
-    fn nested(&mut self) -> ProofTreeFormatter<'_, '_> {
-        ProofTreeFormatter { f: self, on_newline: true }
+impl<'a, 'b> ProofTreeFormatter<'a, 'b> {
+    pub(super) fn new(f: &'a mut (dyn Write + 'b)) -> Self {
+        ProofTreeFormatter { f }
+    }
+
+    fn nested<F, R>(&mut self, func: F) -> R
+    where
+        F: FnOnce(&mut ProofTreeFormatter<'_, '_>) -> R,
+    {
+        func(&mut ProofTreeFormatter { f: &mut Indentor { f: self.f, on_newline: true } })
     }
 
     pub(super) fn format_goal_evaluation(&mut self, goal: &GoalEvaluation<'_>) -> std::fmt::Result {
-        let f = &mut *self.f;
-
         let goal_text = match goal.is_normalizes_to_hack {
             IsNormalizesToHack::Yes => "NORMALIZES-TO HACK GOAL",
             IsNormalizesToHack::No => "GOAL",
         };
 
-        writeln!(f, "{}: {:?}", goal_text, goal.uncanonicalized_goal,)?;
-        writeln!(f, "CANONICALIZED: {:?}", goal.canonicalized_goal)?;
+        writeln!(self.f, "{}: {:?}", goal_text, goal.uncanonicalized_goal)?;
+        writeln!(self.f, "CANONICALIZED: {:?}", goal.canonicalized_goal)?;
 
         match &goal.kind {
             GoalEvaluationKind::CacheHit(CacheHit::Global) => {
-                writeln!(f, "GLOBAL CACHE HIT: {:?}", goal.result)
+                writeln!(self.f, "GLOBAL CACHE HIT: {:?}", goal.result)
             }
             GoalEvaluationKind::CacheHit(CacheHit::Provisional) => {
-                writeln!(f, "PROVISIONAL CACHE HIT: {:?}", goal.result)
+                writeln!(self.f, "PROVISIONAL CACHE HIT: {:?}", goal.result)
             }
             GoalEvaluationKind::Uncached { revisions } => {
                 for (n, step) in revisions.iter().enumerate() {
-                    let f = &mut *self.f;
-                    writeln!(f, "REVISION {n}: {:?}", step.result)?;
-                    let mut f = self.nested();
-                    f.format_evaluation_step(step)?;
+                    writeln!(self.f, "REVISION {n}: {:?}", step.result)?;
+                    self.nested(|this| this.format_evaluation_step(step))?;
                 }
-
-                let f = &mut *self.f;
-                writeln!(f, "RESULT: {:?}", goal.result)
+                writeln!(self.f, "RESULT: {:?}", goal.result)
             }
         }?;
 
         if goal.returned_goals.len() > 0 {
-            let f = &mut *self.f;
-            writeln!(f, "NESTED GOALS ADDED TO CALLER: [")?;
-            let mut f = self.nested();
-            for goal in goal.returned_goals.iter() {
-                writeln!(f, "ADDED GOAL: {:?},", goal)?;
-            }
+            writeln!(self.f, "NESTED GOALS ADDED TO CALLER: [")?;
+            self.nested(|this| {
+                for goal in goal.returned_goals.iter() {
+                    writeln!(this.f, "ADDED GOAL: {:?},", goal)?;
+                }
+                Ok(())
+            })?;
+
             writeln!(self.f, "]")?;
         }
 
@@ -72,58 +83,53 @@ impl ProofTreeFormatter<'_, '_> {
         &mut self,
         evaluation_step: &GoalEvaluationStep<'_>,
     ) -> std::fmt::Result {
-        let f = &mut *self.f;
-        writeln!(f, "INSTANTIATED: {:?}", evaluation_step.instantiated_goal)?;
+        writeln!(self.f, "INSTANTIATED: {:?}", evaluation_step.instantiated_goal)?;
 
         for candidate in &evaluation_step.candidates {
-            let mut f = self.nested();
-            f.format_candidate(candidate)?;
+            self.nested(|this| this.format_candidate(candidate))?;
         }
-        for nested_goal_evaluation in &evaluation_step.nested_goal_evaluations {
-            let mut f = self.nested();
-            f.format_nested_goal_evaluation(nested_goal_evaluation)?;
+        for nested in &evaluation_step.nested_goal_evaluations {
+            self.nested(|this| this.format_nested_goal_evaluation(nested))?;
         }
 
         Ok(())
     }
 
     pub(super) fn format_candidate(&mut self, candidate: &GoalCandidate<'_>) -> std::fmt::Result {
-        let f = &mut *self.f;
-
         match &candidate.kind {
             CandidateKind::NormalizedSelfTyAssembly => {
-                writeln!(f, "NORMALIZING SELF TY FOR ASSEMBLY:")
+                writeln!(self.f, "NORMALIZING SELF TY FOR ASSEMBLY:")
             }
             CandidateKind::Candidate { name, result } => {
-                writeln!(f, "CANDIDATE {}: {:?}", name, result)
+                writeln!(self.f, "CANDIDATE {}: {:?}", name, result)
             }
         }?;
 
-        let mut f = self.nested();
-        for candidate in &candidate.candidates {
-            f.format_candidate(candidate)?;
-        }
-        for nested_evaluations in &candidate.nested_goal_evaluations {
-            f.format_nested_goal_evaluation(nested_evaluations)?;
-        }
-
-        Ok(())
+        self.nested(|this| {
+            for candidate in &candidate.candidates {
+                this.format_candidate(candidate)?;
+            }
+            for nested in &candidate.nested_goal_evaluations {
+                this.format_nested_goal_evaluation(nested)?;
+            }
+            Ok(())
+        })
     }
 
     pub(super) fn format_nested_goal_evaluation(
         &mut self,
         nested_goal_evaluation: &AddedGoalsEvaluation<'_>,
     ) -> std::fmt::Result {
-        let f = &mut *self.f;
-        writeln!(f, "TRY_EVALUATE_ADDED_GOALS: {:?}", nested_goal_evaluation.result)?;
+        writeln!(self.f, "TRY_EVALUATE_ADDED_GOALS: {:?}", nested_goal_evaluation.result)?;
 
         for (n, revision) in nested_goal_evaluation.evaluations.iter().enumerate() {
-            let f = &mut *self.f;
-            writeln!(f, "REVISION {n}")?;
-            let mut f = self.nested();
-            for goal_evaluation in revision {
-                f.format_goal_evaluation(goal_evaluation)?;
-            }
+            writeln!(self.f, "REVISION {n}")?;
+            self.nested(|this| {
+                for goal_evaluation in revision {
+                    this.format_goal_evaluation(goal_evaluation)?;
+                }
+                Ok(())
+            })?;
         }
 
         Ok(())

--- a/compiler/rustc_middle/src/traits/solve/inspect/format.rs
+++ b/compiler/rustc_middle/src/traits/solve/inspect/format.rs
@@ -4,7 +4,7 @@ pub(super) struct ProofTreeFormatter<'a, 'b> {
     f: &'a mut (dyn Write + 'b),
 }
 
-/// A formatter which adds 4 spaces of identation to its input before
+/// A formatter which adds 4 spaces of indentation to its input before
 /// passing it on to its nested formatter.
 ///
 /// We can use this for arbitrary levels of indentation by nesting it.

